scrapper.py

```diff
@@ -26,38 +36,133 @@ class Crawler:
     Crawler implementation
     """
     def __init__(self, seed_urls, max_articles: int):
-        pass
+        self.seed_urls = seed_urls
+        self.max_articles = max_articles
+        self.urls = []
 
     def _extract_url(self, article_bs):
-        pass
+        class_bs = article_bs.find('div', class_='view-content view-rows')
+        title_bs = class_bs.find_all('td', class_="views-field views-field-title table__cell")
+        for link_bs in title_bs:
+            if len(self.urls) >= self.max_articles:
+                break
+            link_bs = link_bs.find('a')
+            self.urls.append('https://rjano.ruslang.ru' + link_bs['href'])
+
 
     def find_articles(self):
         """
         Finds articles
         """
-        pass
+        headers = {'user-agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) '
+                                 'AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.102 Safari/537.36'}
+        for seed_url in self.seed_urls:
+            response = requests.get(seed_url, headers)
+            article_bs = BeautifulSoup(response.text, 'html.parser')
+            self._extract_url(article_bs)
 
     def get_search_urls(self):
         """
         Returns seed_urls param
         """
-        pass
+        return self.seed_urls
 
 
 def prepare_environment(base_path):
     """
     Creates ASSETS_PATH folder if not created and removes existing folder
     """
-    pass
+    if os.path.isdir(base_path):
+        files = os.listdir(base_path)
+        for file in files:
+            os.remove(os.path.join(base_path,file))
+    else:
+        os.makedirs(base_path)
 
 
 def validate_config(crawler_path):
     """
     Validates given config
     """
-    pass
+    with open(crawler_path, 'r', encoding='utf-8') as file:
+        config = json.load(file)
+    seed_urls = config['seed_urls']
+    max_articles = config['total_articles_to_find_and_parse']
+    for seed_url in seed_urls:
+        if not re.match(r'https?://', seed_url):
+            raise IncorrectURLError
+    if not seed_urls:
+        raise IncorrectURLError
+    if not isinstance(seed_urls, list):
+        raise IncorrectURLError
+    if not isinstance(max_articles, int):
+        raise IncorrectNumberOfArticlesError
+    if max_articles > 100:
+        raise NumberOfArticlesOutOfRangeError
+    if max_articles == 0 or max_articles < 0:
+        raise IncorrectNumberOfArticlesError
+    prepare_environment(ASSETS_PATH)
```

Comment:
it cannot be performed in a function that *validates*

Reply:
prepare_environment is not called in the tests, but without tmp/articles tests won't pass. What should I do with it?

Reply:
really?

1. move this call to a more appropriate place
2. check tests one more time. report back if they really fail